tests/tracing/test_sampling.py

```diff
@@ -284,3 +284,61 @@ def test_warns_and_sets_sampled_to_false_on_invalid_traces_sampler_return_value(
         transaction = start_transaction(name="dogpark")
         logger.warning.assert_any_call(StringContaining("Given sample rate is invalid"))
         assert transaction.sampled is False
+
+
+@pytest.mark.parametrize(
+    "traces_sample_rate,sampled_output,reports_output",
+    [
+        (None, False, []),
+        (0.0, False, [("sample_rate", "transaction")]),
+        (1.0, True, []),
+    ],
+)
+def test_records_lost_event_only_if_traces_sample_rate_enabled(
+    sentry_init, traces_sample_rate, sampled_output, reports_output, monkeypatch
+):
+    reports = []
+
+    def record_lost_event(reason, data_category=None, item=None):
+        reports.append((reason, data_category))
+
+    sentry_init(traces_sample_rate=traces_sample_rate)
+
+    monkeypatch.setattr(
+        Hub.current.client.transport, "record_lost_event", record_lost_event
+    )
```

Comment:
I'd probably use a custom transport instead of `monkeypatch` because doing so tends to be less brittle.

I was about to disregard this, but then noticed that all other tests in the same file use `mock` and `mock.patch` for the same effect -- any reason to do it differently here? If not, then let's at least keep local consistency.

Reply:
I followed the same style as all other client report related tests from Armin.
Examples:
https://github.com/getsentry/sentry-python/blob/b681fcfbbdaf8cae55f069d97b1e310dbb657d60/tests/test_transport.py#L198
https://github.com/getsentry/sentry-python/blob/b681fcfbbdaf8cae55f069d97b1e310dbb657d60/tests/test_basics.py#L103-L105

Reply:
The local consistency I brought up comes from ideas as in https://www.python.org/dev/peps/pep-0008/#a-foolish-consistency-is-the-hobgoblin-of-little-minds.

Considering that the different test files are following different styles already, I'll let this be your call.